client/src/main/java/org/eclipse/hono/client/HonoClient.java

```diff
@@ -68,6 +68,18 @@
      */
     Future<HonoClient> connect();
 
+    /**
+     * Disconnects from the Hono server.
+     */
+    void disconnect();
+
+    /**
+     * Disconnects from the Hono server.
+     *
+     * @param completionHandler The completion handler to invoke with the result of the operation.
```

Comment:
I think it would also be important to write about the fact that - as it is implemented in this PR - none of the "usual" disconnectHandlers are called here. 

With "usual" I mean the `disconnectHandler` param given to the `connect()` method or, if that was not set, the implicit disconnectHandler that invokes `reconnect()` (see `handleConnectionLoss()`).

Reply:
ok thanks. however, i tried very hard to understand how can the `disconnectHandler` local parameter of the `connect()` method be invoked in the `disconnect()` method. if you consider the following usage of the honoclient: `connect()` -> `disconnect()` -> `connect()` -> `disconnect()`. i assume here that a call to `disconnect` does not automatically connects to the server until an explicit call to `connect()` is made.

could you please clarify?

Reply:
The reason why I suggested using the `disconnectHandler` supplied to the `connect` method in the #500 ticket text is that I considered it convenient for the API user to have the disconnect logic  (usually, I guess: reconnect and when done, re-establish AMQP-Links on the new connection) only be defined in one place.

But thinking more about it, I think that might not be intuitive (and more complicated in the HonoClientImpl, having to keep the `disconnectHandler` supplied to the `connect` method in a new HonoClientImpl field). 
That's why I prefer your approach now, letting the user supply a `disconnectHandler` to the `disconnect()` method.

Assuming that in most cases a HonoClient should always be connected and have AMQP-links established on the connection, I think it would be good to point out in the `disconnect` javadoc, that an appropriate "reconnect" logic (if needed) should be put in the `disconnectHandler` param of the `disconnect()` method.